streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/EnvironmentTest.java

```diff
@@ -0,0 +1,176 @@
+
+        EmailSettingForm emailSettingForm =
+            environmentPage.createEnvironment(EnvironmentDetailForm.EnvSettingTypeEnum.Email)
+                .<EmailSettingForm>addSetting(EnvironmentDetailForm.EnvSettingTypeEnum.Email)
+                .host(emailHost)
+                .port(emailPort)
+                .address(emailAddress)
+                .user(emailUser)
+                .password(emailPassword)
+                .ok();
+
+        String expectedErrorMessage =
+            "connect to target mail server failed: 535 Error: authentication failed";
+        Awaitility.await()
+
+            .untilAsserted(
+                () -> {
+                    new WebDriverWait(browser, Constants.DEFAULT_WEBDRIVER_WAIT_DURATION);
+                    assertThat(environmentPage.errorMessageList())
+                        .as("Connect failed error message should be displayed")
+                        .extracting(WebElement::getText)
+                        .anyMatch(it -> it.contains(expectedErrorMessage));
+                });
+
+        new WebDriverWait(browser, Constants.DEFAULT_WEBDRIVER_WAIT_DURATION)
+            .until(ExpectedConditions.elementToBeClickable(environmentPage.errorMessageConfirmButton()));
+        environmentPage.errorMessageConfirmButton().click();
+        emailSettingForm.cancel();
+    }
+
+    @Test
+    @Order(130)
+    public void testCreateDockerSettingFailed() {
+        final EnvironmentPage environmentPage = new EnvironmentPage(browser);
+
+        DockerSettingForm dockerSettingForm =
+            environmentPage.createEnvironment(EnvironmentDetailForm.EnvSettingTypeEnum.Docker)
+                .<DockerSettingForm>addSetting(EnvironmentDetailForm.EnvSettingTypeEnum.Docker)
+                .address(dockerAddress)
+                .namespace(dockerNamespace)
+                .user(dockerUser)
+                .password(dockerPassword)
+                .ok();
+
+        String expectedErrorMessage = String.format(
+            "Failed to validate Docker registry, error: Status 500: {\"message\":\"login attempt to %s failed with status: 404 Not Found\"}",
+            dockerAddress);
+        Awaitility.await()
+
+            .untilAsserted(
+                () -> {
+                    new WebDriverWait(browser, Constants.DEFAULT_WEBDRIVER_WAIT_DURATION);
+                    assertThat(environmentPage.errorMessageList())
+                        .as("Failed to validate docker registry error message should be displayed")
+                        .extracting(WebElement::getText)
+                        .anyMatch(it -> it.contains(expectedErrorMessage));
+                });
+
+        new WebDriverWait(browser, Constants.DEFAULT_WEBDRIVER_WAIT_DURATION)
+            .until(ExpectedConditions.elementToBeClickable(environmentPage.errorMessageConfirmButton()));
+        environmentPage.errorMessageConfirmButton().click();
+        dockerSettingForm.cancel();
+    }
+}
```

Comment:
Can't pass at the same time because `EmailSettingForm#buttonOk()` and `DockerSettingForm#buttonOk()` are unstable?